mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+            queries = self._read_path_queries(path)
+            augmented_queries = self.augment_queries(queries, **kwargs)
+            self._write_files(path, augmented_queries)
+
+    @abstractmethod
+    def augment_queries(self, queries):
+        """Generates augmented data given application queries.
+
+        Args:
+            queries (list): List of queries.
+
+        Return:
+            augmented_queries (list): List of augmented queries.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def _get_files(self, config):
+        """Fetches relevant files given the path rules specified in the config.
+
+        Args:
+            config (dict): Application config.
+
+        Return:
+            filtered_paths (list): List of file paths to be augmeted.
+        """
+        all_file_paths = self._resource_loader.get_all_file_paths()
+
+        if not config["paths"]:
+            logger.warning(
+                """'paths' field is not configured or misconfigured in the `config.py`.
+                 Can't find files to augment."""
+            )
+            return
+
+        rules = config["paths"]
+
+        filtered_paths = []
+
+        for rule in rules:
+            pattern = _get_pattern(rule)
+            compiled_pattern = re.compile(pattern)
+            filtered_paths.extend(
+                self._resource_loader.filter_file_paths(
+                    compiled_pattern=compiled_pattern, file_paths=all_file_paths
+                )
+            )
+        return filtered_paths
+
+    def _read_path_queries(self, path):
+        """Returns all queries in a specified file path.
+
+        Args:
+            path (str): File path.
+
+        Return:
+            queries (list): List of queries in the file path.
+        """
+        with open(path, "r") as f:
+            queries = f.readlines()
+        return queries
+
+    def _write_files(self, path, augmented_queries):
+        """Writes augmented queries to a new file in the path.
+
+        Args:
+            path (str): File path to the original file.
+            augmented_queries (list): List of augmented queries returned by class.
+        """
+        write_path = path.strip(".txt") + ".augmented.txt"
+
+        with open(write_path, "w") as outfile:
+            for query in augmented_queries:
+                outfile.write(query.strip("\n") + "\n")
+
+
+class EnglishParaphraser(Augmentor):
+    """Paraphraser class for generating English paraphrases."""
+
+    def __init__(self, app_path, config=None):
+        """Initializes an English paraphraser.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+
+        model_name = "tuner007/pegasus_paraphrase"
```

Comment:
@vijay120 Should we also have a local copy of these model in an S3 bucket or something as a fallback in case huggingface decides to phase it out for some reason?

Reply:
I agree. @jre21 could you help us to upload the model to one of our local aws buckets?

Reply:
@ritvikshrivastava I can upload the model to the s3 bucket, let me know.

Reply:
Lets do this in a subsequent PR